x/concentrated-liquidity/tick_test.go

```diff
@@ -811,3 +822,108 @@ func (s *KeeperTestSuite) TestValidateTickRangeIsValid() {
 		})
 	}
 }
+
+func (s *KeeperTestSuite) TestGetAllInitializedTicksForPool() {
+	const (
+		// chosen randomly
+		defaultPoolId = 676
+	)
+
+	defaultTick := withPoolId(defaultTick, defaultPoolId)
+
+	tests := []struct {
+		name                   string
+		preSetTicks            []genesis.FullTick
+		expectedTicksOverwrite []genesis.FullTick
+		expectedError          error
+	}{
+		{
+			name:        "one positive tick per pool",
+			preSetTicks: []genesis.FullTick{defaultTick},
+		},
+		{
+			name:        "one negative tick per pool",
+			preSetTicks: []genesis.FullTick{withTickIndex(defaultTick, -1)},
+		},
+		{
+			name:        "one zero tick per pool",
+			preSetTicks: []genesis.FullTick{withTickIndex(defaultTick, 0)},
+		},
+		{
+			name: "multiple ticks per pool",
+			preSetTicks: []genesis.FullTick{
+				defaultTick,
+				withTickIndex(defaultTick, -1),
+				withTickIndex(defaultTick, 0),
+				withTickIndex(defaultTick, -200),
+				withTickIndex(defaultTick, 1000),
+				withTickIndex(defaultTick, -999),
+			},
+			expectedTicksOverwrite: []genesis.FullTick{
+				withTickIndex(defaultTick, -999),
+				withTickIndex(defaultTick, -200),
+				withTickIndex(defaultTick, -1),
+				withTickIndex(defaultTick, 0),
+				defaultTick,
+				withTickIndex(defaultTick, 1000),
+			},
+		},
+		{
+			name: "multiple ticks per multiple pools",
+			preSetTicks: []genesis.FullTick{
+				defaultTick,
+				withTickIndex(defaultTick, -1),
+				withPoolId(withTickIndex(defaultTick, 0), 3),
+				withTickIndex(defaultTick, -200),
+				withTickIndex(defaultTick, 1000),
+				withTickIndex(defaultTick, -999),
+				withPoolId(withTickIndex(defaultTick, -4), 90),
+				withTickIndex(defaultTick, 33),
+				withPoolId(withTickIndex(defaultTick, 44), 1200),
+				withPoolId(withTickIndex(defaultTick, -1000), 3),
+				withTickIndex(defaultTick, -1234),
+				withPoolId(withTickIndex(defaultTick, 1000), 3), // duplicate for another pool.
+			},
+			expectedTicksOverwrite: []genesis.FullTick{
+				withTickIndex(defaultTick, -1234),
+				withTickIndex(defaultTick, -999),
+				withTickIndex(defaultTick, -200),
+				withTickIndex(defaultTick, -1),
+				defaultTick,
+				withTickIndex(defaultTick, 33),
+				withTickIndex(defaultTick, 1000),
+			},
+		},
+	}
+
+	for _, test := range tests {
+		s.Run(test.name, func() {
+			s.Setup()
+
+			for _, tick := range test.preSetTicks {
+				s.App.ConcentratedLiquidityKeeper.SetTickInfo(s.Ctx, tick.PoolId, tick.TickIndex, tick.Info)
+			}
+
+			// If overwrite is not specified, we expect the pre-set ticks to be returned.
+			expectedTicks := test.preSetTicks
+			if len(test.expectedTicksOverwrite) > 0 {
+				expectedTicks = test.expectedTicksOverwrite
+			}
+
+			// System Under Test
+			ticks, err := s.App.ConcentratedLiquidityKeeper.GetAllInitializedTicksForPool(s.Ctx, defaultPoolId)
+
+			if test.expectedError != nil {
+				s.Require().Error(err)
+				s.Require().ErrorContains(err, test.expectedError.Error())
```

Comment:
I don't see any error cases. Can we either add error cases or remove this conditional?

Reply:
I remove the error condition because it is non-trivial to trigger an error case for this function.

Given how extensively the underlying helpers are tested, I think the effort to trigger an error case here is not justified. Let me know if any objections